### PR TITLE
Added log button and Trignometric Inverse Functions

### PR DESCRIPTION
**Pull request by techimanshu8**
I have added log button for both log 2 base 10 and normal log .
And added sin , cos and tan inverse functions . also added a inverse function to calculate 1/X .
I had also included test for these
Screenshot from 2020-09-23 14-49-14

### CODE REVIEW

**Comment by jeremypw** [comment]
Tip:  You can make this a Draft review if you are still working it :smile:

**Comment by techimanshu8** [comment]
> Tip: You can make this a Draft review if you are still working it
@jeremypw  Thanks for suggestion but I have done all , now This is ready for review .

**Comment by techimanshu8** [comment]
Resolves #139

File: src/MainWindow.vala [hunk 1]
```diff
@@ -201,6 +201,15 @@ namespace PantheonCalculator {
...
             var button_e = new Button ("e", _("Euler's Number"));
+            var button_log = new Button ("log", _("log 2 base 10"));
+            var button_ln = new Button ("ln", _("log base e"));
+            var button_asin = new Button ("sin<sup>-1</sup>", _("arc sine"));
+            button_asin.function = "asin";
+            var button_acos = new Button ("cos<sup>-1</sup>", _("arc cosine"));
+            button_acos.function = "acos";
+            var button_atan = new Button ("tan<sup>-1</sup>", _("arc tangent"));
+            button_atan.function = "atan";
+            var button_inverse = new Button ("X<sup>-1</sup>");
```
Comment (jeremypw) [inline src/MainWindow.vala:212]: In line with other button labels this should have a lowercase 'x'.

**Review by jeremypw — CHANGES_REQUESTED** [review]
Thanks for providing this!  It works well and in general looks good.  Just a few minor issues noted in line.

File: src/MainWindow.vala [hunk 2]
```diff
@@ -201,6 +201,15 @@ namespace PantheonCalculator {
             var button_tanh = new Button ("tanh", _("Hyperbolic Tangent"));
             var button_pi = new Button ("π", _("Pi"));
             var button_e = new Button ("e", _("Euler's Number"));
+            var button_log = new Button ("log", _("log 2 base 10"));
+            var button_ln = new Button ("ln", _("log base e"));
+            var button_asin = new Button ("sin<sup>-1</sup>", _("arc sine"));
+            button_asin.function = "asin";
+            var button_acos = new Button ("cos<sup>-1</sup>", _("arc cosine"));
+            button_acos.function = "acos";
+            var button_atan = new Button ("tan<sup>-1</sup>", _("arc tangent"));
```
Comment (jeremypw) [inline src/MainWindow.vala:210]: The tooltips should be in title case:  "Logarithm Base 10", "Natural Logarithm", "Inverse Sine", "Inverse Cosine", "Inverse Tangent".   ("Arcsine", "Arccosine" and "Arctangent" are also OK)

**Comment by jeremypw** [comment]
Another thing you will need to do is update the appdata file in `\data` for the next release. As there is a new functionality make it v 1.6.0.
```
    <release version="1.6.0" date="2020-05-17" urgency="medium">
      <description>
        <p>Improvements:</p>
        <ul>
          <li>Add log and inverse trigonometric functions</li>
        </ul>
      </description>
    </release>
```

**Comment by jeremypw** [comment]
@techimanshu8 Unfortunately this is failing one of the Core tests - could you look into it? Thanks

**Comment by jeremypw** [comment]
Hmm - looks like the faulty line is not due to this PR - it is line 75 in CoreTest.vala.  Not sure how that got into master :disappointed:  For convenience could you fix it here?  It should be: `assert_equal ("10^(log(2.2))", "2.2");` I suppose.

**Comment by jeremypw** [comment]
Ah, I see you changed the definition of "log" for evaluation.  I think maybe the button should be labelled ""log<sub>10</sub>" to make this clear.

**Comment by techimanshu8** [comment]
Ya, Sure  we can change label , but   I had also changed Coretest.vala accordingly but don`t know why it is not considering.

**Comment by jeremypw** [comment]
As far as I can tell, you added some new lines to CoreTest.vala (which are fine) but one of the existing tests started failing because you redefined `log` elsewhere.  Previously `log` was interpreted as log<sub>e</sub>.  Now it is interpreted as log<sub>10</sub>.  So the test which is now on line 75 needs updating as indicated above.

**Review by jeremypw — APPROVED** [review]
Good to go - thanks!

**Pull request merged**